### PR TITLE
this fix #69 : Cannot obtain text for binary file type : UNKNOWN

### LINKED ISSUE #69: Cannot obtain text for binary file type : UNKNOWN
Got this error in the IDEA while in the Go lang project:
```Error while indexing /home/dds/.gopath/src/gopkg.in/yaml.v2/LICENSE.libyaml
To reindex this file IDEA has to be restarted: Cannot obtain text for binary file type : UNKNOWN
com.intellij.util.indexing.FileContentImpl$IllegalDataException: Cannot obtain text for binary file type : UNKNOWN
	at com.intellij.util.indexing.FileContentImpl.getContentAsText(FileContentImpl.java:218)
	at com.intellij.util.indexing.FileContentImpl.getPsiFile(FileContentImpl.java:80)
	at org.zalando.intellij.swagger.index.SwaggerDataIndexer.map(SwaggerDataIndexer.java:39)
	at org.zalando.intellij.swagger.index.SwaggerDataIndexer.map(SwaggerDataIndexer.java:28)
	at com.intellij.util.indexing.MapReduceIndex.update(MapReduceIndex.java:461)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:1631)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:1567)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.freezeFileTypeTemporarilyIn(FileTypeManagerImpl.java:507)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:1528)
	at com.intellij.util.indexing.FileBasedIndexImpl.indexFileContent(FileBasedIndexImpl.java:1513)
	at com.intellij.util.indexing.FileBasedIndexImpl.processRefreshedFile(FileBasedIndexImpl.java:1493)
	at com.intellij.util.indexing.FileBasedIndexProjectHandler.a(FileBasedIndexProjectHandler.java:157)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.a(CacheUpdateRunner.java:235)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1061)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.a(CacheUpdateRunner.java:252)
	at com.intellij.openapi.progress.impl.CoreProgressManager$3.run(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:568)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:519)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:164)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.run(CacheUpdateRunner.java:248)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:309)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```
Swagger plugin version: 0.0.26
Ubuntu 16.10
IntelliJ IDEA 2016.3.3
Build #IU-163.11103.6, built on January 16, 2017
JRE: 1.8.0_112-release-408-b6 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

### PR DESCRIPTION
**Pull request by manuc66**

### CODE REVIEW

**Comment by manuc66** [comment]
- fix the file extension checking (and move that check to class FileDetector)
- ensure that FileContent are not considered as binary before trying to convert to PsiFile
- re-use isSwaggerFile in SwaggerDataIndexer.map(@NotNull FileContent inputData) (remove duplicate code)

File: src/main/java/org/zalando/intellij/swagger/file/FileDetector.java [hunk 1]
```diff
@@ -58,4 +60,8 @@ private boolean hasSwaggerJsonKey(final PsiFile psiFile) {
     public boolean isSwaggerFile(final PsiFile file) {
         return isMainSwaggerJsonFile(file) || isMainSwaggerYamlFile(file);
     }
+
+    public boolean isSwaggerContentCompatible(VirtualFile file) {
+        return FilenameUtils.isExtension(file.getName(), new String[]{"json", "yaml", "yml"});
```
Comment (semonte) [inline src/main/java/org/zalando/intellij/swagger/file/FileDetector.java]: We could create new constants for the types in https://github.com/zalando/intellij-swagger/blob/master/src/main/java/org/zalando/intellij/swagger/file/FileConstants.java and use them

File: src/test/java/org/zalando/intellij/swagger/file/FileDetectorTest.java [hunk 2]
```diff
@@ -72,4 +75,47 @@ public void thatIsSwaggerYamlFileBasedOnFileContent() throws Exception {
...
+    public void thatJsonFileExtensionIsSwaggerContentCompatible() throws Exception {
+        VirtualFile yamlFile = mock(VirtualFile.class);
+
+        when(yamlFile.getName()).thenReturn("file.json");
+
+        assertTrue(fileDetector.isSwaggerContentCompatible(yamlFile));
+    }
+
+    @Test
+    public void thatSimilarileExtensionAreNotSwaggerContentCompatible() throws Exception {
```
Comment (semonte) [inline src/test/java/org/zalando/intellij/swagger/file/FileDetectorTest.java]: `thatSimilarileExtensionAreNotSwaggerContentCompatible` to `thatSimilarFileExtensionsAreNotSwaggerContentCompatible`

**Review by semonte — CHANGES_REQUESTED** [review]
This is looking great, made a few comments. Once they are fixed, I will merge.

**Comment by semonte** [comment]
@manuc66 The following tests fail:

```
org.zalando.intellij.swagger.file.FileDetectorTest > thatIsSwaggerYamlFileBasedOnFileName FAILED
    java.lang.AssertionError at FileDetectorTest.java:45
org.zalando.intellij.swagger.file.FileDetectorTest > thatIsSwaggerJsonFileBasedOnFileName FAILED
    java.lang.AssertionError at FileDetectorTest.java:37
```

After they are fixed I'll merge.

**Comment by semonte** [comment]
@manuc66 Thanks for the fix!

**Pull request merged**